.github/workflows/docker.yml

```diff
@@ -0,0 +1,48 @@
+name: Docker image master branch
+on:
+  push:
+    branches:
+      - master
+jobs:
+  docker-build-push:
+    runs-on: self-hosted
```

Comment:
I think this could run on a free-tier from GH... but maybe more work needed for Docker layer caching.
Looks good to me on self-hosted too.